lib/auth/password_test.go

```diff
@@ -117,14 +117,95 @@ func setupPasswordSuite(t *testing.T) *passwordSuite {
 	return &s
 }
 
-func TestUserNotFound(t *testing.T) {
+func TestCheckPassswordWOToken(t *testing.T) {
 	t.Parallel()
 
+	ctx := context.Background()
+	s := setupPasswordSuite(t)
+
+	u, err := types.NewUser("user1")
+	require.NoError(t, err)
+	_, err = s.a.CreateUser(ctx, u)
+	require.NoError(t, err)
+	require.NoError(t, s.a.UpsertPassword("user1", []byte("supersecret123")))
+
+	assert.NoError(t, s.a.checkPasswordWOToken(ctx, "user1", []byte("supersecret123")))
+
+	err = s.a.checkPasswordWOToken(ctx, "user1", []byte("wrongpassword"))
```

Comment:
I would rather run all of those tests via public APIs. I suppose that would mean setting up a cluster and calling AuthenticateUser (or a similar method).

On the upside you could build single table for most of the new tests here.

Reply:
I moved relevant test cases to `auth_login_test.go`.